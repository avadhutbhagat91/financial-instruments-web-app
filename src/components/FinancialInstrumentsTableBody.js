import { CREDIT, EQUITIES } from '../appConstants.js';

const FinancialInstrumentsTableBody = ({ financialInstruments }) => {
  return (
    <tbody>
      {financialInstruments.map((instrument, index) => {
        return (
          <tr
            key={index}
            className={
              instrument.assetClass === EQUITIES
                ? "equities"
                : instrument.assetClass === CREDIT
                ? "credit"
                : "macro"
            }
          >
            <td>{instrument.ticker}</td>
            <td
              className={
                instrument.price > 0 ? "price-positive" : "price-negative"
              }
            >
              {instrument.price}
            </td>
            <td>{instrument.assetClass}</td>
          </tr>
        );
      })}
    </tbody>
  );
};
export default FinancialInstrumentsTableBody;