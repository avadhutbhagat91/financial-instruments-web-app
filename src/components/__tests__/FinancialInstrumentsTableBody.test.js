import React from 'react';
import { shallow } from 'enzyme';
import FinancialInstrumentsTableBody from '../FinancialInstrumentsTableBody';
import financialInstruments from '../../financialInstruments.json';

const setup = (setupProps = {}) => {
    const defaultProps = {
        financialInstruments: financialInstruments
    }
    const props = { ...defaultProps, ...setupProps};
    const wrapper = shallow(<FinancialInstrumentsTableBody {...props}/>);
    return {
        props,
        wrapper
    }
}

describe('FinancialInstrumentsTableBody', () => {
    test('render', () => {
        const { wrapper } = setup();
        expect(wrapper).toMatchSnapshot();
    });
});