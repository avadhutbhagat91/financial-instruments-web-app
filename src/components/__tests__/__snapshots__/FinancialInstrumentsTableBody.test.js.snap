// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinancialInstrumentsTableBody render 1`] = `
<tbody>
  <tr
    className="credit"
    key="0"
  >
    <td>
      ALPHA
    </td>
    <td
      className="price-negative"
    >
      -3150.67
    </td>
    <td>
      Credit
    </td>
  </tr>
  <tr
    className="equities"
    key="1"
  >
    <td>
      BETA
    </td>
    <td
      className="price-positive"
    >
      3791.37
    </td>
    <td>
      Equities
    </td>
  </tr>
  <tr
    className="equities"
    key="2"
  >
    <td>
      GAMMA
    </td>
    <td
      className="price-positive"
    >
      2299.1
    </td>
    <td>
      Equities
    </td>
  </tr>
  <tr
    className="equities"
    key="3"
  >
    <td>
      DELTA
    </td>
    <td
      className="price-positive"
    >
      3132.66
    </td>
    <td>
      Equities
    </td>
  </tr>
  <tr
    className="credit"
    key="4"
  >
    <td>
      EPSILON
    </td>
    <td
      className="price-negative"
    >
      -1168.46
    </td>
    <td>
      Credit
    </td>
  </tr>
  <tr
    className="credit"
    key="5"
  >
    <td>
      ZETA
    </td>
    <td
      className="price-positive"
    >
      2716.78
    </td>
    <td>
      Credit
    </td>
  </tr>
  <tr
    className="macro"
    key="6"
  >
    <td>
      ETA
    </td>
    <td
      className="price-positive"
    >
      3089.2
    </td>
    <td>
      Macro
    </td>
  </tr>
  <tr
    className="macro"
    key="7"
  >
    <td>
      THETA
    </td>
    <td
      className="price-positive"
    >
      1075.44
    </td>
    <td>
      Macro
    </td>
  </tr>
  <tr
    className="macro"
    key="8"
  >
    <td>
      IOTA
    </td>
    <td
      className="price-positive"
    >
      1096.64
    </td>
    <td>
      Macro
    </td>
  </tr>
  <tr
    className="credit"
    key="9"
  >
    <td>
      KAPPA
    </td>
    <td
      className="price-positive"
    >
      2321.17
    </td>
    <td>
      Credit
    </td>
  </tr>
  <tr
    className="credit"
    key="10"
  >
    <td>
      LAMBDA
    </td>
    <td
      className="price-positive"
    >
      1472.2
    </td>
    <td>
      Credit
    </td>
  </tr>
  <tr
    className="macro"
    key="11"
  >
    <td>
      MU
    </td>
    <td
      className="price-positive"
    >
      2136.64
    </td>
    <td>
      Macro
    </td>
  </tr>
  <tr
    className="macro"
    key="12"
  >
    <td>
      NU
    </td>
    <td
      className="price-positive"
    >
      2744.89
    </td>
    <td>
      Macro
    </td>
  </tr>
  <tr
    className="equities"
    key="13"
  >
    <td>
      OMIKRON
    </td>
    <td
      className="price-positive"
    >
      2735.1
    </td>
    <td>
      Equities
    </td>
  </tr>
  <tr
    className="equities"
    key="14"
  >
    <td>
      SIGMA
    </td>
    <td
      className="price-positive"
    >
      1854.19
    </td>
    <td>
      Equities
    </td>
  </tr>
  <tr
    className="macro"
    key="15"
  >
    <td>
      TAU
    </td>
    <td
      className="price-positive"
    >
      2082.71
    </td>
    <td>
      Macro
    </td>
  </tr>
  <tr
    className="equities"
    key="16"
  >
    <td>
      OMEGA
    </td>
    <td
      className="price-positive"
    >
      2306.35
    </td>
    <td>
      Equities
    </td>
  </tr>
  <tr
    className="credit"
    key="17"
  >
    <td>
      PSI
    </td>
    <td
      className="price-positive"
    >
      2997.78
    </td>
    <td>
      Credit
    </td>
  </tr>
</tbody>
`;
